src/modeler/AutoRest.Swagger/Validation/Core/RuleContext.cs

```diff
@@ -19,12 +19,16 @@ public class RuleContext
         /// Initializes a top level context for rules
         /// </summary>
         /// <param name="root"></param>
-        public RuleContext(ServiceDefinition root, Uri file) : this(null)
+        public RuleContext(ServiceDefinition root, Uri file, ServiceDefinitionMetadata metadata) : this(null)
         {
             this.Root = root;
             this.Value = root;
             this.File = file;
-            PopulateResourceTypes(root);
+            if (metadata.ServiceDefinitionDocumentType == ServiceDefinitionDocumentType.ARM)
```

Comment:
this change breaks CI:
https://github.com/Azure/autorest/blob/master/src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs#L52 fails because `resModels` is `null`. Since that didn't happen before, I bet that it is this guard that causes the `context.ResourceModels` to not be populated...
Now, I see that both this guard *and* the rule say that they are `ARM`, but still, this is kind of the only potential point of failure I could make out.